SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -0,0 +1,153 @@
+//==------- ctor_copy.cpp  - DPC++ ESIMD on-device test --------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for esimd copy constructor.
+
+#include "common.hpp"
+
+using namespace cl::sycl;
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Struct that provide method that let get current test case description and
+// method that let call simd constructor for initializer test case
```

Comment:
```suggestion
// Descriptor class for the case of calling constructor in initializer context
```
As a generic hint - let's describe more "why" instead of "what". And let's avoid duplicating the code logic in the comments.
By doing so, we:

- provide laconic hints for intent (which are required for review and later code updates)
- lesser the possibility we would need to fix the comments - for example, if we add the 3rd method

Reply:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/578/commits/3c4a51f2dd397e3fb56059d085f481041e558a26#diff-e20a89556dfa9ec98f8eec4bea698de3975b780ff24a3bde1ced890af1d16581R21)

Reply:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/578/commits/3c4a51f2dd397e3fb56059d085f481041e558a26#diff-e20a89556dfa9ec98f8eec4bea698de3975b780ff24a3bde1ced890af1d16581R21)